pkg/api/vm/handler.go

```diff
@@ -95,11 +117,24 @@ func (h vmActionHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	rw.WriteHeader(http.StatusNoContent)
 }
 
-func (h *vmActionHandler) doAction(rw http.ResponseWriter, r *http.Request) error {
-	vars := util.EncodeVars(mux.Vars(r))
-	action := vars["action"]
-	namespace := vars["namespace"]
-	name := vars["name"]
+func (h *vmActionHandler) IsMatchedResource(resource subresource.Resource, method string) bool {
+	if resource.Name != subresource.VirtualMachines.Resource || method != http.MethodPost {
```

Comment:
Again, shouldn't we have HTTP method `PUT` for idempotent actions and HTTP method `GET` for listing-actions like `findMigratableNodes`?

Reply:
Same reason in https://github.com/harvester/harvester/pull/5704#discussion_r1622033858.